### PR TITLE
Describe the journey of a config variable

### PR DESCRIPTION
**Pull request by gibizer**

### CODE REVIEW

**Comment by kashyapc** [comment]
Not a technical review, but a quick meta comment: as a newbie to this area, this is an excellent write-up! Thank you for writing this up, Gibi

**Comment by EliadCohen** [comment]
Agree with @kashyapc , this makes a lot of sense to me too

**Review by stuggi — APPROVED** [review]
/lgtm

File: journey_of_a_config_variable.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+cpu_models = Nehalem
+```
+to nova-compute in CI to allow live migration between slightly different
+compute node models.
+
+The config goes through the following journey:
+1. The snippet starts its life in the
+   [.zuul.yaml defining the CI job](https://github.com/openstack-k8s-operators/nova-operator/blob/03ba9b30065bc410df99a45ed3aff3bfb28c43f3/.zuul.yaml#L182-L185)
+2. That config snippet is copied into a .conf file.
+   [by the ansbile code in ci_framework](https://github.com/openstack-k8s-operators/ci-framework/blob/28e8d32147d0c71f4060c3da6fc38090b35d6aa0/roles/edpm_deploy/tasks/main.yml#L41-L44)
```
Comment (odyssey4me) [inline journey_of_a_config_variable.md:18]: ansible is not correctly spelled here

File: journey_of_a_config_variable.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
+# The journey of a config variable in openstack-k8s-operators
+
+Our goal is to pass extra configuration from the CI job definition to the 
+nova-compute process running on the EDPM compute node. E.g. we need to pass
+the following
+```ini=
+[libvirt]]
```
Comment (rebtoor) [inline journey_of_a_config_variable.md:7]: there's an extra `]`

File: journey_of_a_config_variable.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+   the [dataplane-operator translates](https://github.com/openstack-k8s-operators/dataplane-operator/blob/8d8a1051288a065e8fec16e07c907dc08ebbb4d8/controllers/openstackdataplanedeployment_controller.go#L225-L228)
+   the `OpenStackDataPlaneService` CR to 
+   an [`OpenStackAnsibleEE` CR](https://github.com/openstack-k8s-operators/openstack-ansibleee-operator/blob/main/docs/openstack_ansibleee.md)
+   and [ensures](https://github.com/openstack-k8s-operators/dataplane-operator/blob/b8c94dc3631cef634f29760e07b938595a630572/pkg/deployment/deployment.go#L200)
+   that the ConfigMap is translated to an item in the `extraVolumes` field of
+   the `OpenStackAnsibleEE` CR.
+9. The [openstack-ansibleee-operator creates a k8s Job](https://github.com/openstack-k8s-operators/openstack-ansibleee-operator/blob/a0ea58b83c3acc84d450723dda4d356c2555f781/controllers/openstack_ansibleee_controller.go#L204-L207)
+   from the `OpenStackAnsibleEE` CR and instructs k8s to 
+   [mount the ConfigMap to the Pod](https://github.com/openstack-k8s-operators/openstack-ansibleee-operator/blob/main/controllers/openstack_ansibleee_controller.go#L389)
```
Comment (bogdando) [inline journey_of_a_config_variable.md:51]: please use a fixed commit sha reference so that the code would not run away as it changes in the main branch

**Review by fao89 — COMMENTED** [review]
should we merge this?

**Comment by gibizer** [comment]
> should we merge this?

I need to fix up the nits first...

**Pull request merged**